src/OrleansRuntime/MembershipService/MembershipOracle.cs

```diff
@@ -1050,6 +1050,7 @@ private void LogFailedProbe(SiloAddress silo, int pingNumber, Exception failureR
 
                 if (logger.IsVerbose) logger.Verbose("-Going to DeclareDead silo {0} in the table. About to write entry {1}.", entry.SiloAddress.ToLongString(), entry.ToFullString());
                 entry.Status = SiloStatus.Dead;
+                RuntimeClient.Current.BreakOutstandingMessagesToDeadSilo(entry.SiloAddress);
```

Comment:
I see. The decision to put this call here was really rash. The thing is, that local subscribers in the silo that going to declare another silo dead doesn't get notified about that silo death because at [MembershipOracle.cs#L1052](https://github.com/dotnet/orleans/blob/fc0af9079413505de6c5dad928a266fccda74d68/src/OrleansRuntime/MembershipService/MembershipOracle.cs#L1052) the membership entry status  being set to Dead and then at [MembershipOracleData.cs#L243](https://github.com/dotnet/orleans/blob/d66096adab58b1eb33a9c995eb5979134cdbca5d/src/OrleansRuntime/MembershipService/MembershipOracleData.cs#L243) the same object statuses comparison (http://i.imgur.com/RX96jes.png) resulting in skipped call to [Notify subscribers method](https://github.com/dotnet/orleans/blob/d66096adab58b1eb33a9c995eb5979134cdbca5d/src/OrleansRuntime/MembershipService/MembershipOracleData.cs#L226)